app/jobs/hyku_addons/reindex_model_job.rb

```diff
@@ -44,8 +43,11 @@ def mint_doi(work)
         work.update(doi: [register_doi.identifier])
       end
 
-      def can_mint_for?(work)
-        work.doi.present? || work.visibility != "open" || workflow_state(work)&.workflow_state_name != "deposited"
+      def cant_mint_for(work)
+        state = workflow_state(work)
+        cannot_use_workflow_state = ["deposited", nil].exclude?(state&.workflow_state_name)
```

Comment:
Also you have defined another negative here.  Try instead:
````
can_use_workflow_state = ["deposited", nil].include?(state&.workflow_state_name)
````

Then instead call `!can_use_workflow_state`.

Reply:
Thanks PR being closed for such a time as the job is needed again.